### ISSUE TITLE
Missing indentation in some cases when pressing Enter

### ISSUE THREAD

**Issue by winmain** [TiVo/intellij-haxe#492]
**Case 1:** Missing indentation in class body after var or function declaration
```haxe
class Test {
  var a = 5;
| // <- cursor not indented
}
```
Should be
```haxe
class Test {
  var a = 5;
  |
}
```

**Case 2:** Broken indentation for multiline comments
I typing `/**`
```haxe
class Test {
  /**|
}
```
pressing Enter and got this:
```haxe
class Test {
  /**
  * |   // +1 space needed
**/     // +3 spaces needed (1 tab + 1 space)
}
```
The second and third lines looks broken.
How it should look like:
```haxe
class Test {
  /**
   * |
   **/
}
```

Plugin version: `Pre-release candidate 0.10.1-RC1`
Idea: `2016.2`

**Comment by EBatTiVo** [comment]
Actually, how it *should* look depends upon your language preference settings.  However, I believe that none of those settings match the results you are seeing.

I thought @as3boyan fixed this, but maybe we never pulled it because it only worked in v15 and higher, whereas we were hoping to find a v14 solution as well.

**Comment by winmain** [comment]
Thanks for your reply.
Ok, I selected Default scheme for Haxe code style and the problem still exists. The only difference is that there are 4 space indentation in Default scheme instead of 2 in my scheme.
It would be excellent if I can tune this behavior in settings, but I think it is a bug.

**Comment by winmain** [comment]
@EBatTiVo yes, you were right. I have made a fix for case 1, but it only works in v15 and higher.
I will try to further understand this problem.

**Comment by EBatTiVo** [comment]
@winmain And to confound things further... the unit tests for re-alignment work correctly in all versions.

There is at least one other issue regarding this, and probably two or three.  There might be some more info on those reports.

**Comment by winmain** [comment]
I had to copy `AbstractEnterActionTestCase` from the lastest IDEA to make tests pass. Personally, I don't like this decision, but it works.
@EBatTiVo what do you think?

**Comment by EBatTiVo** [comment]
I don't mind back-porting changes like that.  I *definitely* like to add a comment like:
```haxe
    /* Lifted this file from <old_location> for <reason>. */
```
near the top of the file.  I do a similar thing when I grab entire functions from another source repo.  As long as the licensing terms match, we're good.

**Comment by EBatTiVo** [comment]
Part 1 was fixed with commit https://github.com/TiVo/intellij-haxe/commit/342b568a79ef96cdf2aa26eaa0ea8654ae2d36e1.
